designer/server/src/main/scala/pl/touk/nussknacker/ui/processreport/ProcessCounter.scala

```diff
@@ -35,8 +31,9 @@ class ProcessCounter(fragmentRepository: FragmentRepository) {
 
       nodes.flatMap {
         //TODO: this is a bit of a hack. Metric for fragment input is counted in node with fragment occurrence id...
-        case FlatNode(FragmentInputDefinition(id, _, _)) => Map(id -> nodeCountOption(None))
-        //BranchEndData is kind of artifical entity
+        //We want to count it though while testing fragments
```

Comment:
But when you use fragments inside sceanarios you don't have input definition but input usage. Do you?

Reply:
Hmm, I didn't challenge it but I think it should work like you said. In test `"compute counts for scenario with fragment"` it looks like `FragmentInputDefinition` and `FragmentOutputDefinition` are passed explicitly - I am not sure if that happens in NU testing cases (?). 

What I've done here is just a simple if statement to handle `FlatNode(FragmentInputDefinition)` a little later in `case FlatNode(node)` few lines bellow it - in scenario testing this node is skipped in favor of node that represents fragment which is not present in fragment testing.

Reply:
I see, probably this `FragmentInputDefinition` is somewhere in model even after fragment resolution